src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -161,17 +161,24 @@ server::routes_t get_schema_registry_routes(ss::gate& gate, one_shot& es) {
       ss::httpd::schema_registry_json::compatibility_subject_version,
       wrap(gate, es, compatibility_subject_version)});
 
+    routes.routes.emplace_back(server::route_t{
+      ss::httpd::schema_registry_json::get_restart,
+      wrap(gate, es, get_restart)});
+
     return routes;
 }
 
-ss::future<> service::do_start() {
-    if (_is_started) {
+ss::future<> service::do_start(
+  bool skip_topic_creation = false, bool ignore_started = false) {
+    if (!ignore_started && _is_started) {
         co_return;
     }
     auto guard = gate_guard(_gate);
     try {
         co_await configure();
-        co_await create_internal_topic();
+        if (!skip_topic_creation) {
+            co_await create_internal_topic();
+        }
```

Comment:
This is idempotent:
```suggestion
        co_await create_internal_topic();
```

Reply:
also means you could drop the `skip_topic_creation` boolean from the method signature